modules/geo-layers/src/mvt-layer/mvt-layer.ts

```diff
@@ -323,7 +336,7 @@ export default class MVTLayer<ExtraProps extends {} = {}> extends TileLayer<
       return -1;
     }
 
-    const featureIdToHighlight = isHighlighted ? highlightedFeatureId : hoveredFeatureId;
+    const featureIdToHighlight = isHighlighted ? highlightedFeatureId! : hoveredFeatureId!;
```

Comment:
Nit: Exclamation marks should be used sparingly. I would opt for guarding with an if statement in this case rather than sprinkle in type system defeating exclamation marks.

Reply:
I tend to agree. This felt like a good compromise based on @felixpalmer's feedback to keep the scope of this PR to just type changes.